test/dsl/dsl_advanced_model_construction.jl

```diff
@@ -340,10 +340,23 @@ let
         k[1]*a+k[2], X[1] + V[1]*X[2] --> V[2]*W*Y + B*C
     end
 
-    @parameters k[1:3] a B
```

Comment:
I think this one is intentionally different (i.e. giving different `k` lengths, but only using some, should give the same result). However, there should be a comment pointing this out though.

Reply:
Yeah, but they won't be equal any more as the underlying symbolic array variables are no longer considered the same. I can just not test equality if you prefer then?

Reply:
No, in that case all is good